From d4b665cdd41c28dd83ca6b7584ab426801743c5f Mon Sep 17 00:00:00 2001
From: liwei <liwei1234@oppo.com>
Date: Thu, 20 Nov 2025 16:20:45 +0800
Subject: [PATCH 15/15] ANDROID: GKI: oppo add symbols to symbol list

1 function symbol(s) added
  'int __traceiter_android_vh_oom_evaluate_task_bypass(void*, struct task_struct*, struct oom_control*, bool*)'

1 variable symbol(s) added
  'struct tracepoint __tracepoint_android_vh_oom_evaluate_task_bypass'

Bug: 460636204
Bug: 472245484
Change-Id: I42031a3e285df9c43946bffcefd633cadd7f0a97
Signed-off-by: liwei <liwei1234@oppo.com>
(cherry picked from commit 5453f12f6799d8f600197663a7bf7039c2e0dd7e)
---
 gki/aarch64/abi.stg       | 28 ++++++++++++++++++++++++++++
 gki/aarch64/symbols/oplus |  2 ++
 2 files changed, 30 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index e89d3e5bd7ae..f6f23f8d1a4f 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -414771,6 +414771,14 @@ function {
   parameter_id: 0x18bd6530
   parameter_id: 0x33f8a50c
 }
+function {
+  id: 0x9bd481b1
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x1d19a9d5
+  parameter_id: 0x3c692b7e
+  parameter_id: 0x11cfee5a
+}
 function {
   id: 0x9bd49403
   return_type_id: 0x6720d32f
@@ -438050,6 +438058,15 @@ elf_symbol {
   type_id: 0x9b5ac8cd
   full_name: "__traceiter_android_vh_oom_check_panic"
 }
+elf_symbol {
+  id: 0x1ff6ff8b
+  name: "__traceiter_android_vh_oom_evaluate_task_bypass"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x39e206ae
+  type_id: 0x9bd481b1
+  full_name: "__traceiter_android_vh_oom_evaluate_task_bypass"
+}
 elf_symbol {
   id: 0xb3f4a3bd
   name: "__traceiter_android_vh_oom_killer_disable"
@@ -445304,6 +445321,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_oom_check_panic"
 }
+elf_symbol {
+  id: 0x3a3a83c9
+  name: "__tracepoint_android_vh_oom_evaluate_task_bypass"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_oom_evaluate_task_bypass"
+}
 elf_symbol {
   id: 0xeb498f27
   name: "__tracepoint_android_vh_oom_killer_disable"
@@ -518441,6 +518467,7 @@ interface {
   symbol_id: 0x721b87b7
   symbol_id: 0xd3e1b817
   symbol_id: 0x594fa05d
+  symbol_id: 0x1ff6ff8b
   symbol_id: 0xb3f4a3bd
   symbol_id: 0x2884ad94
   symbol_id: 0x724577f8
@@ -519247,6 +519274,7 @@ interface {
   symbol_id: 0xae5bbde9
   symbol_id: 0x508ff939
   symbol_id: 0x0c86f8fb
+  symbol_id: 0x3a3a83c9
   symbol_id: 0xeb498f27
   symbol_id: 0x05b8726a
   symbol_id: 0xdfe6a3ee
diff --git a/gki/aarch64/symbols/oplus b/gki/aarch64/symbols/oplus
index e9c63c9f2bd0..764229bbdf15 100644
--- a/gki/aarch64/symbols/oplus
+++ b/gki/aarch64/symbols/oplus
@@ -345,6 +345,7 @@
   __traceiter_android_vh_mutex_opt_spin_finish
   __traceiter_android_vh_mutex_opt_spin_start
   __traceiter_android_vh_mutex_unlock_slowpath
+  __traceiter_android_vh_oom_evaluate_task_bypass
   __traceiter_android_vh_page_add_new_anon_rmap
   __traceiter_android_vh_percpu_rwsem_wq_add
   __traceiter_android_vh_ra_tuning_max_page
@@ -488,6 +489,7 @@
   __tracepoint_android_vh_mutex_opt_spin_finish
   __tracepoint_android_vh_mutex_opt_spin_start
   __tracepoint_android_vh_mutex_unlock_slowpath
+  __tracepoint_android_vh_oom_evaluate_task_bypass
   __tracepoint_android_vh_page_add_new_anon_rmap
   __tracepoint_android_vh_percpu_rwsem_wq_add
   __tracepoint_android_vh_ra_tuning_max_page
-- 
2.50.1

